[xbmc][fix] Use fmt for logging as well.

Paxxi opened the pull request:
Use fmt for logging as well. Changing the CLog methods to variadic templates.
Fixes a lot of pointer casts that fmt dislikes when using %p.
Removes CWin32Log rather than implementing LogW in the new way.
Hides CLogGlobals in the cpp file, getting rid of criticalsection in header.

Fixes https://trac.kodi.tv/ticket/17565
Likely doesn't build on non-windows yet

garbear commented on xbmc/windowing/rpi/RPIUtils.cpp:
```diff
@@ -560,8 +560,8 @@ void CRPIUtils::GetSupportedModes(HDMI_RES_GROUP_T group, std::vector<RESOLUTION
-      CLog::Log(LOGDEBUG, "EGL mode %d: %s (%.2f) %s%s:%x\n", i, res.strMode.c_str(), res.fPixelRatio,
-          tv->native ? "N" : "", tv->scan_mode ? "I" : "", tv->code);
+      CLog::Log(LOGDEBUG, "EGL mode %d: %s (%.2f) %s%s:%x\n", i, res.strMode, res.fPixelRatio,
+          tv->native ? "N" : "", tv->scan_mode ? "I" : "", int(tv->code));
```
static_cast please

Also removing the .c_str() is compatible with `%s`? Or do we need `{0:s}`-style format?

Paxxi replied:
it's compatible, fmt supports both formats.

Paxxi replied:
Oh and the int() is not a cast, it creates a temporary because the type is a bitfield which fmt can't handle

garbear commented on xbmc/cores/VideoPlayer/DVDCodecs/Video/MMALCodec.cpp:
```diff
@@ -225,10 +230,15 @@ void CMMALVideo::dec_output_port_cb(MMAL_PORT_T *port, MMAL_BUFFER_HEADER_T *buf
+                buffer->type->video.flags, static_cast<void*>(m_pool.get()) omvb->Width(),
+                omvb->Height(), omvb->AlignedWidth(), omvb->AlignedHeight(), m_decoded_width,
+                m_decoded_height, m_decoded_aligned_width, m_decoded_aligned_height,
+                (char*)&omvb->Encoding());
```
Might as well update this one too

Paxxi replied:
Not entirely sure what that cast does, from what I can tell Encoding returns a uint32_t so casting to (char*) is weird.

I've changed things sporadically as I've fixed log statements that uses `%p` so there's a lot of things I haven't touched that should be fixed later on.

Rechi commented:
wrong revision was build
jenkins build this please

Paxxi commented:
@rechi I think this is ready for review.

Paxxi committed:
Move log to use fmt

Fix a bunch of errors around pointers

Get rid of CWin32Log

Refactor CLog a bit, hide CLogGlobals in the cpp file

Add non-variadic overloads for a tiny optimization

Standardize on using LogF in most place. This was found when running sourcetrail and getting clang errors.

FernetMenta commented:
this change makes pvr crash, in particular: https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp#L208

Paxxi commented:
Thanks! fritsch mentioned a crash earlier today but didn't find anything by going through the changes. Fix coming up
